go/vt/discovery/replicationlag_test.go

```diff
@@ -0,0 +1,92 @@
+package discovery
+
+import (
+	"reflect"
+	"testing"
+
+	qpb "github.com/youtube/vitess/go/vt/proto/query"
+)
+
+func TestFilterByReplicationLag(t *testing.T) {
+	// 0 endpoint
+	got := FilterByReplicationLag([]*EndPointStats{})
+	if len(got) != 0 {
+		t.Errorf("FilterByReplicationLag([]) = %+v, want []", got)
+	}
+	// 1 serving endpoint
+	eps1 := &EndPointStats{
+		Serving: true,
+		Stats:   &qpb.RealtimeStats{},
+	}
+	eps2 := &EndPointStats{
+		Serving: false,
+		Stats:   &qpb.RealtimeStats{},
+	}
+	got = FilterByReplicationLag([]*EndPointStats{eps1, eps2})
+	if len(got) != 1 {
+		t.Errorf("FilterByReplicationLag([EPS{Serving: true}, EPS{Serving: false}]) = %+v, want one", got)
+	}
```

Comment:
Might want to set a unique UID in eps, and make sure you get the right result here.